internal/clusterfeature/clusterfeatureadapter/workflow/cluster_feature_apply_activity.go

```diff
@@ -20,28 +20,28 @@ import (
 	"github.com/banzaicloud/pipeline/internal/clusterfeature"
 )
 
-const ClusterFeatureUpdateActivityName = "cluster-feature-update"
+const ClusterFeatureApplyActivityName = "cluster-feature-apply"
 
-type ClusterFeatureUpdateActivityInput struct {
+type ClusterFeatureApplyActivityInput struct {
 	ClusterID   uint
 	FeatureName string
 	FeatureSpec clusterfeature.FeatureSpec
 }
 
-type ClusterFeatureUpdateActivity struct {
-	features clusterfeature.FeatureRegistry
+type ClusterFeatureApplyActivity struct {
+	features clusterfeature.FeatureOperatorRegistry
 }
 
-func MakeClusterFeatureUpdateActivity(features clusterfeature.FeatureRegistry) ClusterFeatureUpdateActivity {
-	return ClusterFeatureUpdateActivity{
+func MakeClusterFeatureApplyActivity(features clusterfeature.FeatureOperatorRegistry) ClusterFeatureApplyActivity {
```

Comment:
```suggestion
func MakeClusterFeatureApplyActivity(features clusterfeature.FeatureOperatorRegistry) *ClusterFeatureApplyActivity {
```

Reply:
I think that this is unnecessary. As a performance optimization it would be premature. As for instance sharing, you could make that decision on the caller's side by taking the address of the returned instance and passing that around. Also, instance sharing is only relevant if you mutate the instance, which this implementation does not do. See also: https://stackoverflow.com/a/23551970/578530